swagger.yaml

```
@@ -5909,6 +5909,8 @@ definitions:
         type: boolean
       removable:
         type: boolean
+      has_partitions:
+        type: boolean
```

Maybe a string is better - you can pass more information to the user.  For example "The disk has a 4 partitions" or "The disk has 3 logical volumes", etc.  Either that, or change the boolean to an int and return only data about partitions.

```suggestion
      formatting:
        type: string
```

---

It's not only about the number; it's about what it is.  The point is that a boolean conveys limited information, so it's usually not a good idea to use it in an API.  If we want to provide the user with more information, we would need to add another property or deprecate this one and replace it.  So, it is better to use a string, and then we can provide the user with more information.  For example, if we told them there were LVs on it, they might pause and check it out.

---

ok but the formatting is kinda bad naming cause we have a field for `format` 
can we call it `partitionTableType` which will accept `[ basic , lvm, none]` and `none` value will be there passsing value for the validation

---

How about `partitionTypes` that returns a string that is a comma-separated list of partition types?  So the user can see if the partitions are LVM, regular linux file systems, RAID, Ceph OSD, etc.